StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1603DocumentationMustContainValidXml.cs

```diff
@@ -45,7 +50,103 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(HandleXmlElement, SyntaxKind.XmlElement);
+            context.RegisterSyntaxNodeAction(HandleXmlEmptyElement, SyntaxKind.XmlEmptyElement);
+            context.RegisterSyntaxNodeAction(HandleXmlCDataSection, SyntaxKind.XmlCDataSection);
+        }
+
+        private void HandleXmlElement(SyntaxNodeAnalysisContext context)
+        {
+            var xmlElementSyntax = context.Node as XmlElementSyntax;
+
+            if (xmlElementSyntax != null)
+            {
+                if (xmlElementSyntax.StartTag.LessThanToken.IsMissing)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.StartTag.GetLocation(), "Xml element start tag is missing a '<'."));
+                }
+
+                if (xmlElementSyntax.StartTag.GreaterThanToken.IsMissing)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.StartTag.GetLocation(), "Xml element start tag is missing a '>'."));
+                }
+
+                if (xmlElementSyntax.EndTag.LessThanSlashToken.IsMissing)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.EndTag.GetLocation(), "Xml element end tag is missing a '</'."));
+                }
+
+                if (xmlElementSyntax.EndTag.GreaterThanToken.IsMissing)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.EndTag.GetLocation(), "Xml element end tag is missing a '>'."));
+                }
+                if (xmlElementSyntax.StartTag.Name.ToString() != xmlElementSyntax.EndTag.Name.ToString())
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.StartTag.GetLocation(), new[] { xmlElementSyntax.EndTag.GetLocation() }, $"The '{xmlElementSyntax.StartTag.Name}' start tag does not match the end tag of '{xmlElementSyntax.EndTag.Name}'."));
```

Comment:
For this particular error, I would report the warning on the end tag with the following message:

> The closing XML tag '{xmlElementSyntax.EndTag.Name}' is mismatched.